### PR TITLE
Support Mongo for SAML2 SP metadata

### PR DESCRIPTION
**Pull request by mmoayyed**
This PR adds `SAML2MongoMetadataGenerator` to the pac4j SAML2 portfolio. This feature is only available to pac4j-SAML that is based on OpenSAML v5. 

Surrounding components have been re-worked and refactored to support the change. Mongo dependencies are also made as optional, and need to be re-declared by users when used. 

Typical use case:

```java
var configuration = new SAML2Configuration();
...
configuration.setMetadataGenerator(new SAML2MongoMetadataGenerator(this.mongoClient));
....
configuration.init();
```

### CODE REVIEW

**Comment by mmoayyed** [comment]
Docs and release notes now updated.

File: pac4j-saml-opensamlv5/pom.xml [hunk 1]
```diff
@@ -173,6 +181,20 @@
...
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <groupId>org.mongodb</groupId>
+            <artifactId>bson</artifactId>
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <groupId>org.mongodb</groupId>
+            <artifactId>mongodb-driver-core</artifactId>
```
Comment (leleuj) [inline pac4j-saml-opensamlv5/pom.xml:196]: Why is this dependency not `optional`?

Comment (leleuj) [reply pac4j-saml-opensamlv5/pom.xml:196]: I thought that this dependency should be explicitly added by the pac4j user.

**Comment by leleuj** [comment]
LGTM. One question though about the `mongodb-driver-core` dependency...

File: pac4j-saml-opensamlv5/pom.xml (hunk 1 above)
Comment (mmoayyed) [reply pac4j-saml-opensamlv5/pom.xml:196]: You're right. Should be fixed and optional now.

**Comment by mmoayyed** [comment]
I will also work on support for JPA in this area, and I think that should be enough for now. Then, will build support for both options in CAS v7.

**Comment by leleuj** [comment]
OK. Good. Merging

**Pull request merged**